sycl/test-e2e/Graph/Explicit/dotp.cpp

```diff
@@ -0,0 +1,85 @@
+// REQUIRES: level_zero, gpu
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// Tests constructing a graph using the explicit API to perform a dotp
+// operation using USM memory.
+
+#include "../graph_common.hpp"
+
+int main() {
+  queue Queue{gpu_selector_v};
+
+  exp_ext::command_graph Graph{Queue.get_context(), Queue.get_device()};
+
+  float *Dotp = malloc_device<float>(1, Queue);
+
+  const size_t N = 10;
+  float *X = malloc_device<float>(N, Queue);
+  float *Y = malloc_device<float>(N, Queue);
+  float *Z = malloc_device<float>(N, Queue);
+
+  auto NodeI = Graph.add([&](handler &CGH) {
+    CGH.parallel_for(N, [=](id<1> it) {
+      const size_t i = it[0];
+      X[i] = 1.0f;
+      Y[i] = 2.0f;
+      Z[i] = 3.0f;
+    });
+  });
+
+  auto NodeA = Graph.add(
+      [&](handler &CGH) {
+        CGH.parallel_for(range<1>{N}, [=](id<1> it) {
+          const size_t i = it[0];
+          X[i] = Alpha * X[i] + Beta * Y[i];
+        });
+      },
+      {exp_ext::property::node::depends_on(NodeI)});
+
+  auto NodeB = Graph.add(
+      [&](handler &CGH) {
+        CGH.parallel_for(range<1>{N}, [=](id<1> it) {
+          const size_t i = it[0];
+          Z[i] = Gamma * Z[i] + Beta * Y[i];
+        });
+      },
+      {exp_ext::property::node::depends_on(NodeI)});
+
+  auto NodeC = Graph.add(
+      [&](handler &CGH) {
+#ifdef TEST_GRAPH_REDUCTIONS
```

Comment:
Is this fallback still needed? If so we it might be better to extract that behavior into an extra test that intentionally fails. At least we should document the macro.

Reply:
Our reduction support is still inconsistent - https://github.com/reble/llvm/issues/159 
I can't think of a good place to document this macro, so I've used the fallback path in the existing tests and then created new tests with a reduction version. More thorough reduction testing could be added when #159 is actioned.